src/libtriton/format/pe/pe.cpp

```diff
@@ -0,0 +1,226 @@
+          }
+
+          area.setOffset(rawAddr);
+          area.setSize(rawSize);
+          area.setVirtualAddress(virtAddr);
+
+          this->memoryMapping.push_back(area);
+        }
+      }
+
+
+      triton::uint64 Pe::getOffsetFromAddress(triton::uint64 vaddr) const {
+        for (auto &&section : this->header.getSectionHeaders()) {
+          if (vaddr >= section.getVirtualAddress() && vaddr < (section.getVirtualAddress() + section.getRawSize())) {
+            triton::uint64 offset = ((vaddr - section.getVirtualAddress()) + section.getRawAddress());
+            if (offset >= section.getRawAddress()+section.getRawSize()) {
+              std::ostringstream os;
+              os << "Pe::getOffsetFromAddress(): address " << std::hex << vaddr << " out of bounds in the " << section.getName() << " section";
+              throw triton::exceptions::Pe(os.str());
+            }
+            return offset;
+          }
+        }
+        std::ostringstream os;
+        os << "Pe::getOffsetFromAddress(): address " << std::hex << vaddr << " not found in any section";
+        throw triton::exceptions::Pe(os.str());
+      }
+
+      void Pe::initExportTable(void) {
+          triton::uint32 exportStart = header.getDataDirectory().getExportTable_rva();
+          triton::uint32 exportSize = header.getDataDirectory().getExportTable_size();
+
+          if (exportStart == 0) {
+              return;   //no export table, leave it blank
+          }
+
+          exportTable.parse(raw+getOffsetFromAddress(exportStart));
+          exportTable.setName((char*)raw+getOffsetFromAddress(exportTable.getNameRVA()));
+          triton::uint32 addrTableStart = getOffsetFromAddress(exportTable.getExportAddressTableRVA());
+
+          std::vector<PeExportEntry> entries;
+          for (triton::usize i=0; i<exportTable.getAddressTableEntries(); ++i) {
+              PeExportEntry entry;
+              triton::uint32 exportRVA;
+              std::memcpy(&exportRVA, raw+addrTableStart+4*i,sizeof(exportRVA));
+              if (exportRVA >= exportStart && exportRVA < exportStart+exportSize) {
+                  entry.isForward = true;
+                  entry.forwarderRVA = exportRVA;
+                  entry.forwarderName = std::string((char*)raw+getOffsetFromAddress(exportRVA));
+              } else {
+                  entry.isForward = false;
+                  entry.exportRVA = exportRVA;
+              }
+              entries.push_back(std::move(entry));
+          }
+
+          triton::uint32 nameTableStart = getOffsetFromAddress(exportTable.getNamePointerRVA());
+          triton::uint32 ordTableStart = getOffsetFromAddress(exportTable.getOrdinalTableRVA());
+          for (triton::usize i=0;i<exportTable.getNumberOfNamePointers(); ++i) {
+              triton::uint16 ordinal;
+              std::memcpy(&ordinal, raw+ordTableStart+2*i,sizeof(ordinal));
```

Comment:
What about if ``exportTable.getNumberOfNamePointers()`` is corrupted (Big number) ?